processor/src/main/java/org/mapstruct/ap/model/IterableMappingMethod.java

```diff
@@ -116,6 +117,10 @@ public IterableMappingMethod build() {
                 false
             );
 
+            if ( method instanceof ForgedMethod ) {
+                method.getThrownTypes().addAll( assignment.getThrownTypes() );
```

Comment:
For most other Methods, `getThrownTypes()` returns an immutable list. How about making the different requirement in `ForgedMethod` explicit by adding a method `addThrownTypes(..)` to `ForgedMethod`? It could also contain the contains-check that was implemented below for the MapMappingMethod builder.
The contains-check could also be avoided by declaring `thrownTypes` as `Set` (using LinkedHashSet as implementation, which guarantees an iteration in the order of the inserts).